tendermint/src/hash.rs

```diff
@@ -25,6 +27,28 @@ pub enum Hash {
     Sha256([u8; SHA256_HASH_SIZE]),
 }
 
+impl DomainType<Vec<u8>> for Hash {}
+
+/// Default conversion from Vec<u8> is SHA256 Hash
+impl TryFrom<Vec<u8>> for Hash {
+    type Error = Error;
+
+    fn try_from(value: Vec<u8>) -> Result<Self, Self::Error> {
+        Hash::new(Algorithm::Sha256, &value)
+    }
+}
+
+impl From<Hash> for Vec<u8> {
+    fn from(value: Hash) -> Self {
+        if value == Hash::default() {
+            return vec![];
+        }
+        match value {
+            Hash::Sha256(s) => s.to_vec(),
+        }
+    }
+}
+
 impl Hash {
     #[allow(clippy::new_ret_no_self)]
```

Comment:
I know this is not strictly part of this PR, but perhaps we could tackle this while we're at it and rename `Hash::new` to `Hash::from_bytes` like Clippy suggests and remove this attribute?

Reply:
Absolutely! Covered in the next commit.